src/app/resources/resources.component.html

```diff
@@ -1,9 +1,21 @@
 <!-- FOR DEMO ONLY -->
 <!-- use the async pipe when fetching for database -->
-<button mat-raised-button routerLink="add" color="primary" routerLinkActive="active" i18n>Add Resource</button>
+<div>
+  <button mat-raised-button routerLink="add" color="primary" routerLinkActive="active" i18n>Add Resource</button>
+  <span class="search" [ngStyle]="{'float': 'right'}">
+    <mat-icon>search</mat-icon>
+    <mat-form-field class="font-size-1">
+      <input matInput [(ngModel)]="searchResource" i18n-placeholder placeholder="Type text to search...">
+    </mat-form-field></span>
+</div>
+
+<div class="mat-paginator-container" [ngStyle]="{'justify-content': 'flex-start'}">
+ <mat-paginator [pageSize]="pageSize" [pageSizeOptions]="pageSizeOptions" (page)="pageEvent = $event"></mat-paginator>
+</div>
+
 <ng-template ngFor let-item [ngForOf]="resources">
   <!-- Used ng-template insted of spans or divs because I did not want any to add any element -->
-  <div class="list-item">
+  <div class="list-item" *ngIf="item.doc.title.toLowerCase().includes(searchResource.toLowerCase())" >
```

Comment:
material has its own filter for filtering data from record array so use that

Reply:
well I m not able to use material features as filter is use ur mat-table..as it has not been use in resource.